client/src/telethon/_impl/client/client/net.py

```diff
@@ -265,7 +270,10 @@ async def invoke_request(
 async def step_sender(client: Client) -> None:
     try:
         assert client._sender
-        updates = await client._sender.step()
+        updates = await client._sender.step_updates()
+    except ConnectionResetError:
+        await connect(client, reconnect=True)
```

Comment:
Retrying forever is not good.

I agree we should have some form of autoreconnect. But I don't think it should be in this PR. It's already been hard to merge so far, no need to introduce other complicated features on top.